Generic-Binding-Lib/Additions/Example.GenericType.cs

```diff
@@ -8,96 +8,209 @@
+		[global::System.ComponentModel.EditorBrowsable (global::System.ComponentModel.EditorBrowsableState.Never)]
+		public override global::Java.Interop.JniPeerMembers JniPeerMembers {
+			get { return global::jniabi.example.GenericType._class; }
+		}
+
+		[global::System.Diagnostics.DebuggerBrowsable (global::System.Diagnostics.DebuggerBrowsableState.Never)]
+		[global::System.ComponentModel.EditorBrowsable (global::System.ComponentModel.EditorBrowsableState.Never)]
+		protected override IntPtr ThresholdClass {
+			get { return global::jniabi.example.GenericType._class.JniPeerType.PeerReference.Handle; }
+		}
+
+		[global::System.Diagnostics.DebuggerBrowsable (global::System.Diagnostics.DebuggerBrowsableState.Never)]
+		[global::System.ComponentModel.EditorBrowsable (global::System.ComponentModel.EditorBrowsableState.Never)]
+		protected override global::System.Type ThresholdType {
+			get { return typeof (GenericType); }
+		}
+
+		protected GenericType (IntPtr javaReference, JniHandleOwnership transfer)
+			: base (javaReference, transfer)
 		{
-			if (cb_PerformanceMethod_Ljava_lang_Object_ == null)
-				cb_PerformanceMethod_Ljava_lang_Object_ = JNINativeWrapper.CreateDelegate ((_JniMarshal_PPL_V) n_PerformanceMethod_Ljava_lang_Object_);
-			return cb_PerformanceMethod_Ljava_lang_Object_;
 		}
 
-		static void n_PerformanceMethod_Ljava_lang_Object_ (IntPtr jnienv, IntPtr native__this, IntPtr native_p0)
+		// Metadata.xml XPath constructor reference: path="/api/package[@name='example']/class[@name='GenericType']/constructor[@name='GenericType' and count(parameter)=0]"
+		[Register (".ctor", "()V", "")]
+		public unsafe GenericType ()
+			: base (IntPtr.Zero, JniHandleOwnership.DoNotTransfer)
 		{
-			var __this = global::Java.Lang.Object.GetObject<global::Example.IGenericTypeInvoker> (jnienv, native__this, JniHandleOwnership.DoNotTransfer);
-			var p0 = global::Java.Lang.Object.GetObject<global::Java.Lang.Object> (native_p0, JniHandleOwnership.DoNotTransfer);
-			__this.InvokePerformanceMethod (p0);
+			const string __id = "()V";
+
+			if (((global::Java.Lang.Object) this).Handle != IntPtr.Zero)
+				return;
+
+			try {
+				var __r = global::jniabi.example.GenericType._class.InstanceMethods.StartCreateInstance (__id, ((object) this).GetType (), null);
+				SetHandle (__r.Handle, JniHandleOwnership.TransferLocalRef);
+				global::jniabi.example.GenericType._class.InstanceMethods.FinishCreateInstance (__id, this, null);
+			} finally {
+			}
+		}
+
+		// Metadata.xml XPath method reference: path="/api/package[@name='example']/class[@name='GenericType']/method[@name='TestPerformance' and count(parameter)=2 and parameter[1][@type='T'] and parameter[2][@type='int']]"
+		[Register ("TestPerformance", "(Ljava/lang/Object;I)V", "")]
+		public unsafe void TestPerformance (Java.Lang.Object? p0, int p1)
+		{
+			try {
+				global::jniabi.example.GenericType.TestPerformance (this, p0?.PeerReference ?? default, p1);
+			}
+			finally {
+				GC.KeepAlive (p0);
```

Comment:
Is this `KeepAlive` necessary since `global::jniabi.example.GenericType.TestPerformance (...)` is also calling `KeepAlive` on it?

Reply:
We can get rid of the "inner" one, certainly.